lib/hammer_cli_foreman/associating_commands.rb

```diff
@@ -234,10 +234,48 @@ module ProvisioningTemplate
 
       class AddProvisioningTemplateCommand < HammerCLIForeman::AddAssociatedCommand
         associated_resource :provisioning_templates
-        desc _("Associate a provisioning template")
+        desc _("Associate provisioning templates")
+
+        option "--provisioning-template-ids", "PROVISIONING_TEMPLATE_IDS", _("List of provisioning template ids"),
+               format: HammerCLI::Options::Normalizers::List.new
+        option "--provisioning-templates", "PROVISIONING_TEMPLATE_NAMES", _("List of provisioning template names"),
+               format: HammerCLI::Options::Normalizers::List.new
```

Comment:
I'd suggest to add here `attribute_name: :option_provisioning_template_names` as well, so we can slightly modify [AddAssociatedCommand command](https://github.com/theforeman/hammer-cli-foreman/blob/f163dd9fc9a232f711621f81a16752624edd5de7/lib/hammer_cli_foreman/commands.rb#L668) like so:

```ruby
    def get_new_ids
      ids = get_current_ids.map(&:to_s)
      required_ids = get_associated_identifiers.map(&:to_s)
      required_ids << get_associated_identifier.to_s

      ids += required_ids
      ids.uniq
    end
```

Sure, we'll need to [add](https://github.com/theforeman/hammer-cli-foreman/blob/f163dd9fc9a232f711621f81a16752624edd5de7/lib/hammer_cli_foreman/commands.rb#L617) a new method for retrieving ids:

```ruby
    def get_associated_identifiers
      get_resource_ids(associated_resource, :scoped => true)
    end
```